packages/next-swc/crates/core/src/react_server_components.rs

```diff
@@ -0,0 +1,338 @@
+use serde::Deserialize;
+
+use swc_core::{
+    common::{
+        comments::{Comment, CommentKind, Comments},
+        errors::HANDLER,
+        FileName, Span, DUMMY_SP,
+    },
+    ecma::ast::*,
+    ecma::utils::{prepend_stmts, quote_ident, quote_str, ExprFactory},
+    ecma::visit::{as_folder, noop_visit_mut_type, Fold, VisitMut, VisitMutWith},
+};
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(untagged)]
+pub enum Config {
+    All(bool),
+    WithOptions(Options),
+}
+
+impl Config {
+    pub fn truthy(&self) -> bool {
+        match self {
+            Config::All(b) => *b,
+            Config::WithOptions(_) => true,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Options {
+    pub is_server: bool,
+}
+
+struct ReactServerComponents<'a, C: Comments> {
```

Comment:
Seems like these fields only stores `&'static str`s, so you can remove `'a` from this struct and change fields to `Vec<&'static str>`